src/agent/src/rpc.rs

```diff
@@ -537,6 +554,54 @@ impl AgentService {
             }
         }
     }
+
+    // When being passed an image name through a container annotation, merge its
+    // corresponding bundle OCI specification into the passed container creation one.
+    async fn merge_bundle_oci(&self, container_oci: &mut oci::Spec) -> Result<()> {
+        if let Some(image_name) = container_oci
+            .annotations
+            .get(&ANNO_K8S_IMAGE_NAME.to_string())
+        {
+            if let Some(container_id) = self.sandbox.clone().lock().await.images.get(image_name) {
+                let mut image_oci_config_path = PathBuf::from(CONTAINER_BASE);
+                image_oci_config_path.push(&container_id);
+                image_oci_config_path.push(&String::from("config.json"));
+                debug!(
+                    sl!(),
+                    "Image bundle config path: {:?}", image_oci_config_path
+                );
+
+                let image_oci =
+                    oci::Spec::load(image_oci_config_path.to_str().ok_or_else(|| {
+                        anyhow!(
+                            "Invalid container image OCI config path {:?}",
+                            image_oci_config_path
+                        )
+                    })?)
+                    .context("load image bundle")?;
+
+                if let Some(container_root) = container_oci.root.as_mut() {
+                    if let Some(image_root) = image_oci.root.as_ref() {
+                        let mut root_path = PathBuf::from(CONTAINER_BASE);
+                        root_path.push(container_id);
+                        root_path.push(image_root.path.clone());
```

Comment:
You can remove the `mut` if you call `.join()` instead of `.push()`.

Reply:
Ah, good call :) Fixed as well.